### PR TITLE
Polyfill String.prototype.startsWith

### PR DESCRIPTION
**Pull request by samouri**
**summary**
Tiny quality-of-life improvement for AMP Devs.
Polyfills `String.prototype.startsWith` for non esm browsers.

### CODE REVIEW

File: src/polyfills/string-starts-with.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
```
Comment (gmajoulet) [inline src/polyfills/string-starts-with.js:2]: 👀

Comment (samouri) [reply src/polyfills/string-starts-with.js:2]: You wouldn't believe when I started working on this file :)

File: 3p/twitter.js [hunk 2]
```diff
@@ -75,7 +74,7 @@ export function twitter(global, data) {
     } else if (data.timelineSourceType) {
       // Extract properties starting with 'timeline'.
       const timelineData = Object.keys(data)
-        .filter((prop) => startsWith(prop, 'timeline'))
+        .filter((prop) => prop.startsWith('timeline'))
```
Comment (jridgewell) [inline 3p/twitter.js:77]: To be used here, this needs to be included in `3p/polyfills.js`

File: src/polyfills/string-starts-with.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  // eslint-disable-next-line local/no-invalid-this
+  return this.substring(pos, pos + search.length) === search;
+}
+
+/**
+ * Sets the String.startsWith polyfill if it does not exist.
+ * @param {!Window} win
+ */
+export function install(win) {
+  if (!win.Array.prototype.includes) {
```
Comment (jridgewell) [inline src/polyfills/string-starts-with.js:37]:
```suggestion
  if (!win.String.prototype.startsWith) {
```

File: src/polyfills/string-starts-with.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * @param {string} search
+ * @param {number=} rawPos
+ * @return {boolean}
+ * @this {string}
+ */
+function startsWith(search, rawPos) {
+  const pos = rawPos > 0 ? rawPos | 0 : 0;
+  // eslint-disable-next-line local/no-invalid-this
+  return this.substring(pos, pos + search.length) === search;
```
Comment (jridgewell) [inline src/polyfills/string-starts-with.js:29]: Nit:
```suggestion
  return this.substr(pos, search.length) === search;
```

**Review by jridgewell — CHANGES_REQUESTED** [review]

File: src/polyfills/string-starts-with.js (hunk 3 above)
Comment (samouri) [reply src/polyfills/string-starts-with.js:37]: this is what i get for sloppy copy/pasting

File: 3p/twitter.js (hunk 2 above)
Comment (samouri) [reply 3p/twitter.js:77]: Good to know!

File: src/polyfills/string-starts-with.js (hunk 4 above)
Comment (samouri) [reply src/polyfills/string-starts-with.js]: I had never used `substr` before! Cool

**Review by jridgewell — APPROVED** [review]

**Pull request merged**